P5/Utilities/extractegXML.xsl

```diff
@@ -19,65 +48,88 @@
-        <p/>
-      </sourceDesc>
-    </fileDesc>
-  </teiHeader>
-  <text>
-    <body>
-      <p>
-    <xsl:for-each select="key('V',1)">
-      <xsl:variable name="N">
-	<xsl:call-template name="loc"/>
-      </xsl:variable>
-      <xsl:result-document href="valid/{$N}">
-	<xsl:copy-of select="."/>
-      </xsl:result-document>
-      <xsl:text disable-output-escaping="yes">&amp;</xsl:text>
-      <xsl:value-of select="$N"/>
-      <xsl:text>;</xsl:text>
-    </xsl:for-each>
-</p>
-    </body>
-  </text>
-</TEI>
+        <fileDesc>
+          <titleStmt>
+            <title>The title</title>
+          </titleStmt>
+          <editionStmt>
+            <p/>
+          </editionStmt>
+          <publicationStmt>
+            <p/>
+          </publicationStmt>
+          <sourceDesc>
+            <p/>
+          </sourceDesc>
+        </fileDesc>
+      </teiHeader>
+      <text>
+        <body>
+          <p>
+            <xsl:for-each select="key('V',1)">
+              <xsl:variable name="N">
+                <xsl:call-template name="loc"/>
+              </xsl:variable>
+              <xsl:result-document href="valid/{$N}">
+                <xsl:apply-templates select="." mode="copy_egXML_checking_validUntil"/>
+              </xsl:result-document>
+              <xsl:text disable-output-escaping="yes">&amp;</xsl:text>
+              <xsl:value-of select="$N"/>
+              <xsl:text>;</xsl:text>
+            </xsl:for-each>
+          </p>
+        </body>
+      </text>
+    </TEI>
   </xsl:template>
-
+  
   <xsl:template name="loc">
     <xsl:for-each select="ancestor::tei:*|ancestor-or-self::teix:*">
       <xsl:value-of select="name(.)"/>
       <xsl:text>_</xsl:text>
       <xsl:choose>
-	<xsl:when test="@ident">
-	  <xsl:text></xsl:text>
-	  <xsl:value-of select="replace(@ident,':','')"/>
-	  <xsl:text></xsl:text>
-	</xsl:when>
-	<xsl:when test="@xml:id">
-	  <xsl:text></xsl:text>
-	  <xsl:value-of select="replace(@xml:id,':','')"/>
-	  <xsl:text></xsl:text>
-	</xsl:when>
-	<xsl:otherwise>
-	  <xsl:number/>
-	</xsl:otherwise>
+        <xsl:when test="@ident">
+          <xsl:text></xsl:text>
+          <xsl:value-of select="replace(@ident,':','')"/>
+          <xsl:text></xsl:text>
+        </xsl:when>
+        <xsl:when test="@xml:id">
+          <xsl:text></xsl:text>
+          <xsl:value-of select="replace(@xml:id,':','')"/>
+          <xsl:text></xsl:text>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:number/>
+        </xsl:otherwise>
       </xsl:choose>
       <xsl:text>-</xsl:text>
     </xsl:for-each>
-</xsl:template>
-
-</xsl:stylesheet>
\ No newline at end of file
+  </xsl:template>
+  
+  <xsl:template match="@validUntil" mode="copy_egXML_checking_validUntil">
+    <!--
+      First, knock off leading & trailing spaces, which are allowed in the
+      attribute value but may not be in the functions we're calling.
+    -->
+    <xsl:variable name="validUntil_date" select="normalize-space(.)"/>
+    <!--
```

Comment:
It took me some time to understand the magic (aka hack) here and I think it looks good. I would only recommend to add some warning about this particular check `$validUntil_date castable as xs:date` that this refers to the current datatype of the validUntil attribute. I.e., if we are to change it, we need to adjust the test. Hence it would be good to add this notice to att.deprecated.xml as well.

Reply:
Well, seems like overkill to me. But since the error of over-documenting is far less problematic than the error of under-documenting, I will add comments to that effect both here and to att.deprecated.xml.
Will push results into this branch.

Reply:
Done at 17a189797.